node/src/main/scala/coop/rchain/node/runtime/Setup.scala

```diff
@@ -106,6 +106,19 @@ object Setup {
 
       // Block storage
       blockStore <- KeyValueBlockStore(rnodeStoreManager)
+      blockKeyValueStore <- rnodeStoreManager
+                             .store("blocks")
+                             .map(
+                               _.toTypedStore[BlockHash, BlockMessage](
+                                 codecBlockHash,
+                                 codecBlockMessage
+                               )
+                             )
```

Comment:
Block store uses compression for BlockMessage serialization.
https://github.com/rchain/rchain/blob/326155661e4be70cde0bfe4043ae08af66ebeba4/block-storage/src/main/scala/coop/rchain/blockstorage/KeyValueBlockStore.scala#L94-L106

Now that DAG is used to find message by block hash (#3521) we can reuse KeyValueTypedStore directly as a block store.
Please create another PR for that with special scodec for BlockMessage which uses compression as it's now defined in `KeyValueBlockStore`. This serializer should wrap BlockMessage protobuf serialization (as you did with `codecBlockMessage`) with addition for compression.